datastream-server/src/main/java/com/linkedin/datastream/server/zk/ZkAdapter.java

```diff
@@ -1046,6 +1062,135 @@ public int cleanUpOrphanConnectorTasks(boolean cleanUpOrphanTasksInConnector) {
     return orphanCount;
   }
 
+  private Map<String, Set<String>> getDatastreamTaskNamesGroupedByConnector() {
+    return getAllAssignedDatastreamTasks()
+        .values()
+        .stream()
+        .flatMap(Collection::stream)
+        .collect(groupingBy(DatastreamTask::getConnectorType, mapping(DatastreamTask::getDatastreamTaskName, toSet())));
+  }
+
+  /**
+   * Identify orphan connector task locks for which there is no corresponding connector task node present and schedule
+   * the clean up after the debounce time.
+   * Lock cleanup must be scheduled after a debounce timer because:
+   *
+   * a. There is no easy way to know if the task is still running even though it is [un/re]assigned.
+   * b. All tasks which are unassigned must be stopped within a fixed amount of time which is <= debounce timer.
+   *
+   * Thus waiting for the debounce timer gives the guarantee that reassigned/dead tasks have actually stopped.
+   *
+   * NOTE: this should be called after the valid tasks have been reassigned or become safe to discard per
+   * strategy requirement.
+   *
+   * This is a costly operation which involves getting all children of /cluster/connectors/lock from Zookeeper. So,
+   * it should be called only once the leader gets elected and has finished the assignment and cleaned up dead tasks.
+   * @param cleanUpOrphanTaskLocksInConnector whether orphan task locks should be removed from zookeeper or just
+   *                                      print warning logs.
+   * @return total orphan task locks identified/cleaned.
+   */
+  public int cleanUpOrphanConnectorTaskLocks(boolean cleanUpOrphanTaskLocksInConnector) {
+    // do not perform this operation if not a leader or another operation is going on.
+    if (!_isLeader || !_orphanLockCleanupFuture.isDone()) {
+      LOG.info("Skipping cleanUpOrphanConnectorTaskLocks. isLeader: {} lock cleanup scheduled: {}",
+          _isLeader, _orphanLockCleanupFuture.isDone());
+      return 0;
+    }
+
+    LOG.info("cleanUpOrphanConnectorTaskLocks called");
```

Comment:
Can we print the value of `cleanUpOrphanTaskLocksInConnector` to know whether we are just identifying orphans vs. cleaning them up?